templates/flavors/aks-aso/credentials.yaml

```diff
@@ -6,4 +6,4 @@ stringData:
  AZURE_SUBSCRIPTION_ID: ${AZURE_SUBSCRIPTION_ID}
  AZURE_TENANT_ID:       ${AZURE_TENANT_ID}
  AZURE_CLIENT_ID:       ${AZURE_CLIENT_ID_USER_ASSIGNED_IDENTITY}
- AUTH_MODE:             ${ASO_CREDENTIAL_SECRET_MODE:-workloadidentity}
+ AUTH_MODE:             ${ASO_CREDENTIAL_SECRET_MODE:-UserAssignedMSI}
```

Comment:
I don't fully understand this change, was part of a trial and error method to see if `aks-aso` flavored cluster would come up. 
`aks-aso` is still broken. Probing into it now.

Reply:
Using `podidentity` when using identity created by the AKS cluster solved the issue.